TACC/Core-Portal
# Bug/FP-885: Ignore allocation checks to cloud.corral and data.tacc

## Overview: ##

As @jchuahtacc [pointed out](https://github.com/TACC/Core-Portal/pull/351#issuecomment-786235960), we cannot rely on an allocation for Corral hosts (data.tacc.utexas.edu or cloud.corral.tacc.utexas.edu) being present in a user's allocations list. Because of this, instead of checking for a Corral allocation, we whitelist Corral hosts when checking for default allocation.

To verify the user **does** have access to the default storage system if the default storage host is set to Corral (i.e. `_PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEM_DEFAULT` points to a Corral host), we must instead do a health check on the system to confirm validity. This will be achieved via https://github.com/TACC/Core-Portal/pull/359.

## Related Jira tickets: ##

* [FP-885](https://jira.tacc.utexas.edu/browse/FP-885)

## Testing Steps: ##

1. Ensure the following is present for you in `settings_secret.py`:
```
_PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEM_DEFAULT = 'stockyard'
_PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEMS = {
    'stockyard': {
        'name': 'My Data (Work)',
        'description': 'My Data on Stockyard for {username}',
        'site': 'cep',
        'systemId': 'corral.home.{username}',
        'host': 'cloud.corral.tacc.utexas.edu',
        'rootDir': '/work/{tasdir}',
        'port': 2222,
        'icon': None,
    }
```
2. Load any app that you know you have an allocation for that app's execution system host (e.g. RStudio and MATLAB run on stampede2)
3. Confirm the app loads with no warning, or shows a missing allocation for the execution host if you do not have that allocation
4. Confirm you do not see any warning for a missing allocation for Corral.
